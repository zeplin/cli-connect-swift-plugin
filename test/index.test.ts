import Plugin from "../src";

describe("Connected Components - Swift Plugin", () => {
    test("Button.swift snippet creation", async () => {
        const plugin = new Plugin();

        const componentCode = await plugin.process({ path: "test/Button.swift", zeplinNames: [] });

        expect(componentCode).toMatchSnapshot();
    });

    test("Card.swift snippet creation", async () => {
        const plugin = new Plugin();

        const componentCode = await plugin.process({ path: "test/Card.swift", zeplinNames: [] });

        expect(componentCode).toMatchSnapshot();
    });
});
