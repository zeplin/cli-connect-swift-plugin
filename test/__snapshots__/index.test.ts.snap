// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Connected Components - Swift Plugin Button.swift snippet creation 1`] = `
Object {
  "description": "# For a painless usage of \`Button\` in an iOS project! üëØ‚Äç‚ôÇÔ∏è

You can apply *italic*, **bold**, or \`code\` inline styles.

## Unordered Lists

- Lists are great,
- but perhaps don't nest
- Sub-list formatting

  - isn't the best.

## Ordered Lists

1. Ordered lists, too
2. for things that are sorted;
3. Arabic numerals
4. are the only kind supported.",
  "lang": "swift",
  "snippet": "Button.init(title: String,
            kind: Kind,
            icon: UIImage?)

Button.init(frame: NSRect)",
}
`;

exports[`Connected Components - Swift Plugin Card.swift snippet creation 1`] = `
Object {
  "description": "üí≥ This is not a gambling card! This is a cool, useful tech card!",
  "lang": "swift",
  "snippet": "Card.init(title: String,
          details: String,
          imageHeader: String?,
          imageSubHeader: String?,
          image: UIImage?,
          imagePosition: ImagePosition,
          size: Size)",
}
`;
